bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -1072,15 +1072,19 @@ public IteratorValue getIterator() {
     public void setLength(long length) {
         handleFrozenArrayValue();
         int newLength = (int) length;
-        rangeCheck(length, size);
         checkFixedLength(length);
+        rangeCheck(length, size);
         fillerValueCheck(newLength, size);
         resizeInternalArray(newLength);
         fillValues(newLength);
         size = newLength;
     }
 
     private void checkFixedLength(long length) {
+        if (arrayType != null && arrayType.getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
Not all tuples should fail right? Shouldn't we be able to remove all from a tuple with only a rest field?

```
[int...] i = [2, 3];
i.removeAll(); // this is valid?
```

Reply:
My understanding is that in above example [2,3] is a tuple value of type [int, int] and removing/setlength changes the type/shape of the value, IMO is illegal.

Reply:
The type is still `[int...]` right? The shape is `[int, int]`. Updates that change the shape can be done as long as they don't violate the inherent type (given the value is not immutable).

IMO, `[int...]` is the same as `int[]`.

Reply:
Create a spec issue to clarify https://github.com/ballerina-platform/ballerina-spec/issues/309